extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java

```diff
@@ -0,0 +1,582 @@
+
+        // Next call to HTTP Source returns a valid response.
+        var body = FAKER.internet().uuid();
+        httpSourceClientAndServer
+                .when(
+                        getRequest(),
+                        once()
+                )
+                .respond(
+                        withResponse(HttpStatusCode.OK_200, body)
+                );
+
+        // HTTP Sink Request & Response
+        httpSinkClientAndServer
+                .when(
+                        postRequest(body),
+                        once()
+                )
+                .respond(
+                        withResponse()
+                );
+
+        // Act & Assert
+        // Initiate transfer
+        initiateTransfer(transferRequestPayload(processId));
+
+        // Wait for transfer to be completed.
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                fetchTransferState(processId)
+        );
+        // Verify HTTP Source server called exactly 3 times.
+        httpSourceClientAndServer.verify(
+                getRequest(),
+                VerificationTimes.exactly(3)
+        );
+        // Verify HTTP Sink server called exactly once.
+        httpSinkClientAndServer.verify(
+                postRequest(body),
+                VerificationTimes.once()
+        );
+    }
+
+    /**
+     * Validate DPF doesn't retry to deliver the data if the sink drops the connection.
+     */
+    @Test
+    public void transfer_sinkDropsConnection_noRetry() {
+        // Arrange
+        // HTTP Source Request & Response
+        var body = FAKER.internet().uuid();
+        var processId = FAKER.internet().uuid();
+        httpSourceClientAndServer
+                .when(
+                        getRequest(),
+                        once()
+                )
+                .respond(
+                        withResponse(HttpStatusCode.OK_200, body)
+                );
+
+        // HTTP sink drops the connection.
+        httpSinkClientAndServer
+                .when(
+                        postRequest(body),
+                        once()
+                )
+                .error(
+                        withDropConnection()
+                );
+
+        // Act & Assert
+        // Initiate transfer
+        initiateTransfer(transferRequestPayload(processId));
+
+        // Wait for transfer to be completed.
+        await().atMost(30, SECONDS).untilAsserted(() ->
+                fetchTransferState(processId)
+        );
+
+        // Verify HTTP Source server called exactly once.
+        httpSourceClientAndServer.verify(
+                getRequest(),
+                VerificationTimes.once()
+        );
+
+        // Verify HTTP Sink server called exactly once.
+        // TODO: reading the code, it is expected that the sink is only
+        //  called once, but it's actually called twice, we need to figure out why that is.
+        //  When the sink returns an error, it is only called once.
```

Comment:
@jimmarino, @paullatzelsperger is the current behavior that the sink is called twice in this scenario as expected? If so, could you please quickly explain why?

Reply:
In general, it depends based on the value of ResponseStatus. Is the dropped connection a fatal error or retryable. 

@cpeeyush Can you describe the scenario being tested?

Reply:
I'm OK with removing the comment and merging when the behavior is as expected.

Reply:
@jimmarino Here in a dropped connection scenario situations is that mock sink server will drop the connection without returning any response. 

And as we see in the code HttpSink client does not have a retry policy configured while posting data that's why we were looking for the reason that why sink mock server is called twice here after server drops connection.

Doing more research I found that it is happening because okhttp client has retryOnConnectionFailure=true by default and that is causing this retry on this connection dropping from server. So as per the current [okhttp client config](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/65d204f865937686b5c3149b37ab2c36a3368cc0/core/base/src/main/java/org/eclipse/dataspaceconnector/core/CoreServicesExtension.java#L138-L142) it is an expected behavior. I have added a comment in test explaining this.